packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map.ts

```diff
@@ -0,0 +1,770 @@
+import { Construct } from 'constructs';
+import { MapBase, BaseMapProps } from './map-base';
+import * as iam from '../../../aws-iam';
+import { IBucket } from '../../../aws-s3';
+import { Aws } from '../../../core';
+import { FieldUtils } from '../fields';
+import { StateGraph } from '../state-graph';
+import { StateMachineType } from '../state-machine';
+import { CatchProps, IChainable, INextable, ProcessorConfig, ProcessorMode, RetryProps } from '../types';
+
+const DISTRIBUTED_MAP_SYMBOL = Symbol.for('@aws-cdk/aws-stepfunctions.DistributedMap');
+
+/**
+ * Base interface for Item Reader configurations
+ */
+export interface IItemReader {
+  /**
+   * S3 Bucket containing objects to iterate over or a file with a list to iterate over
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * Limits the number of items passed to the Distributed Map state
+   *
+   * @default - Distribute Map state will iterate over all items provided by the ItemReader
+   */
+  readonly maxItems?: number;
+
+  /**
+   * Render the ItemReader as JSON object
+   */
+  render(): any;
+
+  /**
+   * Compile policy statements to provide relevent permissions to the state machine
+   */
+  providePolicyStatements(): iam.PolicyStatement[];
+}
+
+/**
+ * Base interface for Item Reader configuration properties
+ */
+interface ItemReaderProps {
```

Comment:
why are we not exporting this?

Reply:
Didn't feel it was necessary as other item props are implementing this interface

Reply:
Miss on my end